mlir/unittests/Analysis/Presburger/SimplexTest.cpp

```diff
@@ -383,4 +383,17 @@ TEST(SimplexTest, addInequality_already_redundant) {
   EXPECT_TRUE(simplex.isMarkedRedundant(1));
 }
 
+TEST(SimplexTest, isRedundant) {
+  Simplex simplex(2);
+  simplex.addInequality({0, -1, 1}); // [0]: y <= 1.
+  simplex.addInequality({1, 0, -1}); // [1]: x >= 1.
+  simplex.addEquality({-1, 1, 0});   // [2]: y = x.
+
+  EXPECT_TRUE(simplex.isRedundant({-1, 0, 2})); // x <= 2.
+  EXPECT_TRUE(simplex.isRedundant({0, 1, 0}));  // y >= 0.
+
+  EXPECT_FALSE(simplex.isRedundant({-1, 0, -1})); // x <= -1.
+  EXPECT_FALSE(simplex.isRedundant({0, 1, -2}));  // y >= 2.
+}
+
 } // namespace mlir
```

Comment:
Can you add a test where the new inequality doesn't make the set empty?

Reply:
I was not 100% sure what you meant here. I decided to change the test polytope x >= 0, y <= 2, y = x and added the check for y >= 1, which is, in this case, a cut inequality and not redundant while not making the set empty.